### PR TITLE
UI: Move warnings to bottom of settings window

### PR DESCRIPTION
**Pull request by GPattenden**
### Description
Add a new QVBoxLayout to OBSBasicSettings.ui to show the warnings in the output advanced recording settings.

**Old layout for warnings in advanced recording settings:**
AdvancedOldSettings

**New layout for warnings in advanced recording settings:**
AdvancedNewSettings

### Motivation and Context
This PR fixes isssue #9276
It also improves the UI so the warnings shown in the advanced Output Recording Settings are consistent with the ones shown in the simple Output tab.

**Simple Output settings (old and new):**
SimpleOutputSettings

### How Has This Been Tested?

Tests are conducted by manually entering MPEG-4 (.mp4) in the recording format and resizing the settings window.

**Environment:**

- windows 11 Version 10.0.22631 Build 22631

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/forms/OBSBasicSettings.ui [hunk 1]
```diff
@@ -3908,6 +3908,22 @@
...
+                       <number>10</number>
+                      </property>
+                      <property name="rightMargin">
+                       <number>10</number>
+                      </property>
+                      <property name="bottomMargin">
+                       <number>10</number>
+                      </property>
+                     </layout>
+                    </item>
```
Comment (Warchamp7) [inline UI/forms/OBSBasicSettings.ui:3926]: Can you remove these margin values?

**Review by Warchamp7 — CHANGES_REQUESTED** [review]
Outside of the margins on the new QVBoxLayout this looks good to me

**Comment by GPattenden** [comment]
@Warchamp7 I've commited the changes. Is everything ok or does it need any modifications?

**Comment by Warchamp7** [comment]
> @Warchamp7 I've commited the changes. Is everything ok or does it need any modifications?

Taking a deeper look, did you hand edit this XML or did you use Qt Creator? That element structure does not look valid to me.

**Comment by Warchamp7** [comment]
I adjusted the PR to use a widget with a layout. Technically still hand edited only this change after making the adjustment in Qt Designer so as not to affect #10419 too heavily

**Comment by GPattenden** [comment]
I had hand edited the XML file. Thank you

**Comment by noah9444** [comment]
I tried Nightly Build.
The warning is still displayed even after changing to "Custom Output (FFmpeg)".

https://github.com/obsproject/obs-studio/assets/37494132/ee5b28a0-aae9-4507-9578-9a60c63daf52

**Comment by RytoEX** [comment]
@noah9444 This PR didn't remove the warning. It moved its location.

**Comment by Warchamp7** [comment]
> @noah9444 This PR didn't remove the warning. It moved its location.

When set to Custom Output you're no longer using stream encoder. It looks like we don't properly remove that warning from the warning list when switching, but because it was inside the Standard mode, it wasn't visible.

This is a bug that this PR has exposed and should get it's own issue @noah9444

**Pull request merged**